packages/cdk8s/lib/api-object.ts

```diff
@@ -121,7 +122,7 @@ export class ApiObject extends Construct {
 
     // convert to "pure data" so, for example, when we convert to yaml these
     // references are not converted to anchors.
-    return JSON.parse(JSON.stringify(removeEmpty(resolve(this, data))));
+    return JSON.parse(stringify(removeEmpty(resolve(this, data))));
```

Comment:
Doing this at the `apiObject` level instead of `Chart` level. I think it makes more sense to do it here.

That being said... I just found that `YAML` version `1.8.0` now supports the flag `sortMapEntries` (https://github.com/eemeli/yaml/releases/tag/v1.8.0)... I'll take a look there as well

Reply:
I like it that we are doing this at the JSON-level inside `ApiObject` and not when we convert to YAML.

Reply:
Awesome! I'l keep it here then.